moveit_planners/ompl/ompl_interface/src/model_based_planning_context.cpp

```diff
@@ -767,72 +767,46 @@ const moveit_msgs::MoveItErrorCodes ompl_interface::ModelBasedPlanningContext::s
 
   moveit_msgs::MoveItErrorCodes result;
   result.val = moveit_msgs::MoveItErrorCodes::FAILURE;
+  ob::PlannerTerminationCondition ptc = constructPlannerTerminationCondition(timeout, start);
+  registerTerminationCondition(ptc);
   if (count <= 1 || multi_query_planning_enabled_)  // multi-query planners should always run in single instances
   {
     ROS_DEBUG_NAMED(LOGNAME, "%s: Solving the planning problem once...", name_.c_str());
-    ob::PlannerTerminationCondition ptc = constructPlannerTerminationCondition(timeout, start);
-    registerTerminationCondition(ptc);
     result.val = errorCode(ompl_simple_setup_->solve(ptc));
     last_plan_time_ = ompl_simple_setup_->getLastPlanComputationTime();
-    unregisterTerminationCondition();
   }
   else
   {
     ROS_DEBUG_NAMED(LOGNAME, "%s: Solving the planning problem %u times...", name_.c_str(), count);
     ompl_parallel_plan_.clearHybridizationPaths();
-    if (count <= max_planning_threads_)
-    {
+
+    auto plan_parallel = [this, &ptc](unsigned int num_planners) {
```

Comment:
nit

```suggestion
    auto plan_parallel = [this, &ptc](unsigned int num_planners)
    {
```

Reply:
Interestingly clang-format didn't complain about this!?

Reply:
Actually clang-format would revert this suggestion!